Simplenote/Classes/SPNoteEditorViewController.m

```diff
@@ -969,8 +989,8 @@ - (void)keyboardWillChangeFrame:(NSNotification *)notification {
         [UIView animateWithDuration:duration animations:animations];
     }
     
-    [self setVisibleRightBarButtonsForEditingMode:isEditing];
-    _keyboardHeight = visibleHeight;
+    self.keyboardHeight = newKeyboardHeight;
+    self.keyboardVisible = newKeyboardHeight > 0;
```

Comment:
Same as [L963](https://github.com/Automattic/simplenote-ios/pull/750/files#diff-92ac1b7e27f673e8c8f34b3969f54fe1R963)

Reply:
Yes, we could set it based on isEditing from above.

I have it in my head not to conflate the keyboard being visible with us being in editing mode though, as that isn't always the case. So for me this feels better for clarity and doesn't hurt readability.

I'm happy to change it if you feel it's unnecessary though.

Reply:
Apologies, I wasn't really clear. What I meant to say is: the check is duplicated. So instead of doing this:

```
self.keyboardVisible = newKeyboardHeight > 0;
```

You can simply do this:

```
self.keyboardVisible = isEditing;
```

Reply:
I understand what you mean here; that the test on line 993 is the same as the one on 963. My concern is that this is only true in the context of this method, which is why I'd rather not conflate the two states generally. To me, this is just a readability issue though.

What I've done is renamed the BOOL on line 963 to `isKeyboardVisible`, then made the change on line 993 to match.

Let me know if I'm making sense here! 😅